components/check-buttons/controller.tsx

```diff
@@ -0,0 +1,54 @@
+import { useState } from "react";
+
+import CheckButtonsView from "components/check-buttons/view";
+
+import { SelectOption } from "types/utils";
+
+interface CheckButtonsProps {
+  options: SelectOption[];
+  multiple?: boolean;
+  onClick: (value: SelectOption | SelectOption[]) => void;
+}
+
+export default function CheckButtons({
+  options,
+  multiple,
+  onClick,
+}: CheckButtonsProps) {
+  const [selected, setSelected] = useState<SelectOption[]>([]);
```

Comment:
instead of maintaining an array of selected options, you can simply keep a map of `{[index: number]: boolean}` and then by clicking on the option you just toggle the boolean from false to true. after that, you can return the maing of selected on the `onClick` function:

```ts
const [selected, setSelected] = useState<{[index: number]: boolean}>({});

// ...

function toggleOption(index: number) {
  setSelected(previous => ({...previous, [index]: !previous[index] });
  onClick(options.map((opt,i) => ({...opt, selected: selected[i}));
}